### PR TITLE
fix: assign globals to `self`

### PR DESCRIPTION
**Pull request by ascorbic**
### Summary

Next.js middleware and edge functions define `self` as a global-like, function-scoped object. However some of Next's built-in polyfill stubs expect `self` to have browser globals. These polyfills aren't supposed to be needed in edge runtime, but Next automatically includes them if a site has certain dependencies (e.g. `unfetch`). This means that for those sites, edge functions were failing because those globals were missing.

This PR adds globals to `self`, meaning they're available if the polyfill needs them.

### Test plan

This PR adds `isomorphic-unfetch` to the middleware demo site, to simulate this bug. To test it, visit [the middleware demo and load `/shows/222`](https://deploy-preview-1823--next-plugin-edge-middleware.netlify.app/shows/222). The response headers should include `x-example-server`.

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal
capy and puppies

Fixes https://github.com/netlify/pod-ecosystem-frameworks/issues/315

### Standard checks:

- [ ] Check the Deploy Preview's Demo site for your PR's functionality
- [ ] Add docs when necessary

---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.

### CODE REVIEW

**Review by ericapisani — CHANGES_REQUESTED** [review]
Small thing.

I also tried tested in a local environment (`ntl dev`) and ran into a `TypeError`:

`TypeError: this._headers.getAll is not a function`

**Comment by ascorbic** [comment]
Was that with `framework=#static`? Because if not then it's not using this code. It sounds like we need to add a getAll polyfill to our dev function too

File: packages/runtime/src/helpers/edge.ts [hunk 1]
```diff
@@ -118,6 +116,9 @@ const fetch = async (url, init) => {
   }
 }
 
+// Next edge runtime uses "self" as a function-scoped global-like object, but some of the older polyfills expect it to equal globalThis
+const self = { ...globalThis, fetch }
```
[the remark this answers is not in the document]
Comment (ascorbic) [reply packages/runtime/src/helpers/edge.ts:121]: Added a link to [the docs](https://nextjs.org/docs/basic-features/supported-browsers-features#polyfills)

**Comment by ericapisani** [comment]
@ascorbic It wasn't when I originally posted my comment but trying it again with `framework=#static` still results in the same error

**Pull request merged**